From 3fcc81ad127c267018aed96a359bbf01db1bd8ee Mon Sep 17 00:00:00 2001
From: Saul Wold <sgw@linux.intel.com>
Date: Sat, 1 Apr 2017 11:43:31 -0700
Subject: [PATCH] zc: Force 4.10 get_user_pages_remote API

This change forces the newer API, as the linux-intel 4.9-LTS kernel
backported the 4.10 API change in the page handling code.

Signed-off-by: Saul Wold <sgw@linux.intel.com>
---
 zc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zc.c b/zc.c
index ae464ff..b2759ce 100644
--- a/zc.c
+++ b/zc.c
@@ -65,7 +65,7 @@ int __get_userbuf(uint8_t __user *addr, uint32_t len, int write,
 #elif (LINUX_VERSION_CODE < KERNEL_VERSION(4, 9, 0))
 	ret = get_user_pages_remote(task, mm,
 			(unsigned long)addr, pgcount, write, 0, pg, NULL);
-#elif (LINUX_VERSION_CODE < KERNEL_VERSION(4, 10, 0))
+#elif (LINUX_VERSION_CODE < KERNEL_VERSION(4, 10, 0) & 0)
 	ret = get_user_pages_remote(task, mm,
 			(unsigned long)addr, pgcount, write ? FOLL_WRITE : 0,
 			pg, NULL);
